lnrpc/rpc.proto

```diff
@@ -800,6 +798,17 @@ message ChannelPoint {
     uint32 output_index = 3 [json_name = "output_index"];
 }
 
+message OutPoint {
+    /// Raw bytes representing the transaction id.
+    bytes txid_bytes = 1 [json_name = "txid_bytes"];
+
+    /// Reversed, hex-encoded string representing the transaction id.
+    string txid_str = 2 [json_name = "txid_str"];
```

Comment:
I still have mixed feelings about the redundancy in returning a txid in two formats. What was again the reason for not just sticking to the most efficient format bytes? For lncli is doesn't matter, we can convert locally.

Also if we use OutPoint to pass data into lnd in the future, we need to make sure we always use a helper function to either take the raw bytes or parse the string.

Reply:
Mainly for when it's served over REST. If we don't encode ourselves, then it becomes base58 when one is expecting a hex encoded txid. In certain areas we don't encode raw bytes for ourselves manually, which causes confusion for users when for example, their payment preimage is base 58 encoded.

Reply:
I see. It would be nicer if this was somehow transparently done in the proxy generation. Mark the txid fields with an attribute and in the proxy the conversion to reversed hex will be made.